components/admin_console/admin_sidebar/admin_sidebar.tsx

```diff
@@ -22,55 +22,70 @@ import Highlight from 'components/admin_console/highlight';
 import SearchIcon from 'components/widgets/icons/search_icon';
 import QuickInput from 'components/quick_input';
 
-const renderScrollView = (props) => (
+import {AdminConfig, ClientLicense} from 'mattermost-redux/types/config';
+import {PluginRedux, PluginsResponse} from 'mattermost-redux/types/plugins';
+import {CloudState} from 'mattermost-redux/types/cloud';
+import {ConsoleAccess} from 'mattermost-redux/types/admin';
+
+import AdminDefinition from '../admin_definition';
+
+const renderScrollView = (props: Props) => (
     <div
         {...props}
         className='scrollbar--view'
     />
 );
 
-const renderScrollThumbHorizontal = (props) => (
+const renderScrollThumbHorizontal = (props: Props) => (
     <div
         {...props}
         className='scrollbar--horizontal'
     />
 );
 
-const renderScrollThumbVertical = (props) => (
+const renderScrollThumbVertical = (props: Props) => (
     <div
         {...props}
         className='scrollbar--vertical'
     />
 );
 
-class AdminSidebar extends React.PureComponent {
-    static propTypes = {
-        license: PropTypes.object.isRequired,
-        config: PropTypes.object,
-        plugins: PropTypes.object,
-        adminDefinition: PropTypes.object,
-        cloud: PropTypes.object,
-        buildEnterpriseReady: PropTypes.bool,
-        siteName: PropTypes.string,
-        onFilterChange: PropTypes.func.isRequired,
-        navigationBlocked: PropTypes.bool.isRequired,
-        consoleAccess: PropTypes.object,
-        intl: intlShape.isRequired,
-        showTaskList: PropTypes.bool,
-        actions: PropTypes.shape({
-
-            /*
-             * Function to get installed plugins
-             */
-            getPlugins: PropTypes.func.isRequired,
-        }).isRequired,
-    }
+export type Props = {
+    adminDefinition: typeof AdminDefinition;
+    buildEnterpriseReady: boolean;
+    config: DeepPartial<AdminConfig>;
+    consoleAccess: ConsoleAccess;
+    cloud: CloudState;
+    intl: IntlShape;
+    license: ClientLicense;
+    navigationBlocked: boolean;
+    onFilterChange: (term: string) => void;
+    showTaskList: boolean;
+    plugins?: Record<string, PluginRedux>;
+    siteName?: string;
+    actions: {
+
+        /*
+        * Function to get installed plugins
+        */
+        getPlugins: () => Promise<{data: PluginsResponse}>;
+    };
+}
+
+type State = {
+    sections: string[] | null;
+    filter: string;
+}
+
+class AdminSidebar extends React.PureComponent<Props, State> {
+    searchRef: React.RefObject<HTMLInputElement>;
+    idx: any;
```

Comment:
Unfortunately I couldnt set this type. When `console.log`, I see different types such as `string`, `function` or `object` so I kept as `any`. Please let me know if you have any suggestion.

Reply:
Unfortunately, I couldnt find types from `FlexSearch`. After a quick research, I found this npm package for [types](https://www.npmjs.com/package/@types/flexsearch). However, I saw that `FlexSearch`is only used here, so please let me know if it makes sense to add this package just for this, Otherwise I can try to generate the type by myself.

Reply:
I'd be good with importing the `@types` package, it wouldn't be particularly large and enhances our code base. Go ahead and do that :)

Reply:
I spoke too soon 😄  `@types` package I mentioned earlier was created starting from version `0.7.0` for `FlexSearch` package. Currently Mattermost is using `FlexSearch` with version `0.6.32`. Therefore types doesn't match. 

I order to be consistent, I will generate the type myself in this util [file](https://github.com/mattermost/mattermost-webapp/blob/master/utils/admin_console_index.jsx) as `generateIndex` is created here. This means I will also migrate it (and test) to TypeScript. 

Finally, to close the loop, I would also migrate [this](https://github.com/mattermost/mattermost-webapp/blob/master/utils/admin_console_plugin_index.js)  (and test) to TypeScript as it is closely linked

Reply:
Depending on how complicated the types we need are, I think making our own type definitions would work well enough for the time being. We've already done that for a few other libraries [here](https://github.com/mattermost/mattermost-webapp/tree/master/types/external).

There's a ticket in the backlog to update flexsearch ([link](https://mattermost.atlassian.net/browse/MM-37116)). I can't remember what problems I ran into with doing that, so I don't know how much effort that would be.